dashboard/app/controllers/transfers_controller.rb

```diff
@@ -0,0 +1,128 @@
+class TransfersController < ApplicationController
+  before_filter :authenticate_user!
+
+  # POST /section/:id/transfers
+  def create
+    new_section_code = params[:id]
+
+    begin
+      new_section = Section.find_by!(code: new_section_code)
+    rescue ActiveRecord::RecordNotFound
+      # TODO: i18n
+      render json: {
+        error: "Sorry, but that section does not exist. Please enter a different section code."
+      }, status: :not_found
+      return
+    end
+
+    # This is needed when we eventually allow students to be in multiple sections
+    if params.has_key?(:current_section_code)
+      current_section_code = params[:current_section_code]
+    else
+      # TODO: i18n
+      render json: {
+        error: "Please provide current_section_code."
+      }, status: :bad_request
+      return
+    end
+
+    begin
+      current_section = Section.find_by!(code: current_section_code)
+    rescue ActiveRecord::RecordNotFound
+      # TODO: i18n
+      render json: {
+        error: "Sorry, but section #{current_section_code} does not exist. Please enter a different section code."
+      }, status: :not_found
+      return
+    end
+
+    current_section_belongs_to_logged_in_user = current_user.sections.map(&:code).include?(current_section_code)
+    if !current_section_belongs_to_logged_in_user
+      # TODO: i18n
+      render json: {
+        error: "You cannot move students from a section that does not belong to you."
+      }, status: :forbidden
+      return
+    end
+
+    # As of right now, this only applies to transfers to another teacher
+    # When students are allowed to be in multiple sections, this will also be needed
+    # for transfers between the current logged-in teacher
+    new_section_belongs_to_current_teacher = current_user.sections.map(&:code).include?(new_section_code)
+    if new_section_belongs_to_current_teacher
+      stay_enrolled_in_current_section = false
+    else
+      if params.has_key?(:stay_enrolled_in_current_section)
```

Comment:
when you are implementing the frontend make sure that this value is what you expect (eg. if the frontend passes "0" and "1" it won't automatically parse this into true or false; for checkboxes it's common to signal false by not including the key/value pair at all, which would break this line)